doc/UserGuide.md

```diff
@@ -827,6 +834,23 @@ When implementing a shrinking method, one has to be careful to only return
 loop. ScalaCheck has implicit shrinking methods for common types such as integers
 and lists.
 
+If the generator for a type is restricting the range of valid values by
+construction or using `Gen.suchThat`, the values that fail tests can still be
+shrunk without checking the condition, and then ultimately be reported as
+failing even though they do not satisfy it. To avoid that, use `Shrink.suchThat`
+with the condition to be maintained:
+
+```scala
+/** Generate lists of an even length */
+val genEvenList: Gen[List[Int]] = Gen.sized { size =>
+  Gen.listOfN(size * 2, arbitrary[Int])
+}
+
+/** Shrink a list, maintaining even length */
+val shrinkEvenList: Shrink[List[Int]] =
+  implicitly[Shrink[List[Int]]].suchThat(_.length % 2 == 0)
```

Comment:
Is it possible to make `shrinkEvenList` implicit?  Even if I make it lazy, I get an infinite loop when I use it in a property test.

In your test, you use the explicit shrink version of `forAll`.  Unfortunately, that only works for single arity `forAll`.

Reply:
Please review #644 when you get a chance, thanks.